template/pdk.gen.go.ejs

```diff
@@ -152,22 +149,22 @@ func _<%- goName(ex.name) %>() int32 {
 	// It takes input of <%- toGolangType(imp.input) %> (<%- formatCommentLine(imp.input.description) %>)
 	<% } -%>
 	<% if (hasComment(imp.output)) { -%>
-	// And it returns an output <%- toGolangType(imp.output) %> (<%- formatCommentLine(imp.output.description) %>)
+	// And it returns an output <%- toGolangType(imp.output, false, true) %> (<%- formatCommentLine(imp.output.description) %>)
 	<% } -%>
-	func <%- goName(imp.name) %>(<%- imp.input ? `input ${toGolangType(imp.input)}` : null %>) <%- imp.output ? `(${pointerToGolangType(imp.output)}, error)` : "error" %> {
+	func <%- goName(imp.name) %>(<%- imp.input ? `input ${toGolangType(imp.input)}` : null %>) <%- imp.output ? `(${toGolangType(imp.output, true, true)}, error)` : "error" %> {
 		var err error
 		_ = err
 	<% if (imp.input) { -%>
-		<% if (imp.input.contentType === 'application/json') { -%>
+		<% if (isJsonEncoded(imp.input)) { -%>
 		mem, err := pdk.AllocateJSON(&input)
 		if err != nil {
 			return <%- imp.output ? "nil, err" : "err" %>
 		}
-		<% } else if (imp.input.$ref && imp.input.$ref.enum) { -%>
+		<% } else if (isEnum(imp.input)) { -%>
 		mem := pdk.AllocateString(input.String())
-		<% } else if (imp.input.contentType === 'text/plain; charset=UTF-8') { -%>
+		<% } else if (isUtf8Encoded(imp.input.contentType)) { -%>
```

Comment:
I notice on L186, you use `isUtf8Encoded(imp.output)` vs here you pass the `contentType` -- is that intentional, and it supports these values? or does this just maybe not get tested.

Reply:
hmm no it's a mistake. maybe not getting exercised